### PR TITLE
Enable UI deployment

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

Amends the helm template and Docker build to allow Kubernetes deployments.

## What is the intent behind these changes?

To re-enable deployment to the Cloud Platform.

🙏 Please review commit-by-commit.

### CODE REVIEW

File: helm_deploy/hmpps-interventions-ui/templates/_envs.tpl [hunk 1]
```diff
@@ -11,42 +11,42 @@ env:
     valueFrom:
       secretKeyRef:
         name: hmpps-auth
-        key: interventions-ui-client-id
+        key: interventions-ui-client-id.txt
```
Comment (tomsmyers) [inline helm_deploy/hmpps-interventions-ui/templates/_envs.tpl:14]: this is annoying - something must have gone wrong when i uploaded the secrets.

Comment (sldblog) [reply helm_deploy/hmpps-interventions-ui/templates/_envs.tpl:14]: I can also change it on the namespace side if you'd like?

File: Dockerfile [hunk 2]
```diff
@@ -53,8 +55,6 @@ COPY --from=build --chown=appuser:appgroup \
 COPY --from=build --chown=appuser:appgroup \
         /app/node_modules ./node_modules
 
-RUN npm prune --production
```
Comment (tomsmyers) [inline Dockerfile:56]: what's the reason for moving this ? doesn't this mean that the node_modules on the final image contains the dev dependencies?

File: Dockerfile [hunk 3]
```diff
@@ -34,6 +34,8 @@ RUN CYPRESS_INSTALL_BINARY=0 npm ci --no-audit && npm run build  && \
     export GIT_REF=${GIT_REF} && \
     npm run record-build-info
 
+RUN npm prune --no-audit --production
```
Comment (tomsmyers) [inline Dockerfile:39]: what's the reason for moving this here?

Comment (sldblog) [reply Dockerfile:39]: Sorry, I've added the narrative about this in 9d366b4ab1cb1e4928cf597da9785a70f111ffba:

> Prune production dependencies in the build stage
>
> Removing the packages in the final stage will not result in size savings
> as COPY layer will be retained: we need to remove the unneeded packages
> in the `build` phase so the `COPY --from=build` commands copy less.

Comment (sldblog) [reply Dockerfile:39]: The difference in size between current `main` and this branch's Docker build:
```
$ docker images | grep 'ui-'
ui-before  latest                  6b974cc88063        2 seconds ago       425MB
ui-after   latest                  250473b4e6a7        2 minutes ago       226MB
```

**Review by tomsmyers — APPROVED** [review]

**Pull request merged**